agent/consul/leader.go

```diff
@@ -375,7 +375,7 @@ func (s *Server) revokeLeadership() {
 
 	s.stopConnectLeader()
 
-	s.setCAProvider(nil, nil)
+	s.caManager.setCAProvider(nil, nil)
```

Comment:
Should you reset the rest of the internal state of the CAManager on step-down?

Reply:
I think nil-ing out the provider/root and setting the state back to uninitialized is enough, since that will block anything from getting the provider or doing any operations that aren't InitializeCA (which only gets called on establishLeadership).